### PR TITLE
Interlink 2.0

### PR DESCRIPTION
**Pull request by ghost**
### The shuttle/pods have not been moved in this PR, due to other complications. Consider this a remap of the Interlink itself, the Evac Dock will not be accessible for now.

The predecessor of #10527, rather than simply moving the shuttle to the Interlink...
\
hh

The Interlink gets a full rework :)

Most notable of course is the Evac shuttle docking here (removing EORG), but I've made a lot of other really nice overhauls to the Interlink itself:
- Guide lines on the floor, which are also labelled in the info pamphlets; **Blue for Relay, Green for Dorms, and White for Offices**
- A changing bio-hall each shift; see all of the variants! (The plated area in the bottom-left)
- A little pool (including pool toys)
- A proper Arcade (including 'animatronics' and a prize counter!)
- No more stealing the crew from the Interlink, the cutouts are now unable to be moved.
- ACTUAL guide pamphlets (the ones we used were from some random ruin)
- New fun lorebits! Go learn!
- SolFed Fastpass:tm:, letting First Responders enter/leave the round in a nicer way than handheld-teleporter/deleter
- _"Dead-end" hallways with hazard cones, signalling for future mappers to try building off there first!_

_**To-Do:**_

- [ ] Fix SolFed Fastpass
- [x] UI for SolFed Fastpass
- [ ] SolFed spawns near the lift

## How This Contributes To The Skyrat Roleplay Experience
EORG's removal isn't my decision, nor do I care frankly.
What I DO care about is that it's done in the cleanest way possible; if everyone is going to dock here in the end, it may as well be an enjoyable place to dock AT.
I wont pull the trigger, though. Mostly because of another issue that came up with it that I'd rather sort out. (That being getting the Relay to lock down, which I frankly cant get to work)

Additionally, this gives the chance to clean up 911's spawn/despawn as well, giving them a designated place to go to delete themselves, and a designated spawn as well.

**Also removes our old bench sprites so we can use the Greyscale ones properly :)**

## Changelog

:cl:
expansion: The Interlink's Nanotrasen Wing has finished construction, and your shift will now begin there! Remember to pick up a pamphlet to learn more, and the basics are to follow the Blue line! Glory to the Company!
/:cl:

### CODE REVIEW

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
Very quick review: The dorm rooms don't have bolting buttons nor do they have light switches. I'd *really* like those to be there :)

**Comment by ghost** [comment]
> Very quick review: The dorm rooms don't have bolting buttons nor do they have light switches. I'd _really_ like those to be there :)

fuck I knew I forgot something in there, I'll get 'em

**Review by jjpark-kb — CHANGES_REQUESTED** [review]
Please also include a docking bay for escape pods.

File: modular_skyrat/modules/mapping/code/machinery.dm [hunk 1]
```diff
@@ -28,3 +28,146 @@
...
+/obj/machinery/computer/solgov_gtfo/proc/locate_lift_tiles()
+	for(var/turf/open/floor/plating/elevatorshaft/solgov_gtfo/lift_tile in world)
+		if(!lift_tile || lift_tile.z != src.z)	//If the turf somehow has no location (bugged) or isnt on the same z-level as the console, we skip it
+			continue
+		linked_tiles += lift_tile
+	if(!linked_tiles)	//Shits fucked, SOMEHOW
+		log_game("[src] has no linked tiles to delete users from!")
+		message_admins("[ADMIN_LOOKUPFLW(src)] has no linked tiles to delete users from! Fix this before 911 is called by spawning /turf/open/floor/plating/elevatorshaft/solgov_gtfo around it!")
+
+/obj/item/beamout_tool/attack_self(mob/user, modifiers)
```
Comment (Iamgoofball) [inline modular_skyrat/modules/mapping/code/machinery.dm]: what's this doing here?

File: modular_skyrat/modules/mapping/code/machinery.dm [hunk 2]
```diff
@@ -28,3 +28,146 @@
...
+	if(.)
+		return
+
+	switch(action)
+		if("activate_lift")
+			activate_lift()
+
+///Finds users on the 'lift' gives ample warning, then deletes them all. Effectively, its how 911 will leave the round.
+/obj/machinery/computer/solgov_gtfo/proc/activate_lift(mob/user, list_of_riders)
+	if(user.client.holder)	//Debug purposes, admins get a slightly modified warning specifying 'hey, admin, dont use this if you arent debugging'
```
Comment (Iamgoofball) [inline modular_skyrat/modules/mapping/code/machinery.dm]: you should also check if they're a ghost

**Comment by restlessObserver** [comment]
Are you sure that the players entering the round will not spawn in the dorm rooms? It would certainly be funny but awkward

File: modular_skyrat/modules/mapping/code/machinery.dm (hunk 1 above)
Comment (ghost) [reply modular_skyrat/modules/mapping/code/machinery.dm]: Reference while I get it all working

**Comment by ghost** [comment]
> Are you sure that the players entering the round will not spawn in the dorm rooms? It would certainly be funny but awkward

The blue X's are latejoin spawns, which all are at the tram; its close to the Relay now, and randomly appearing in the middle of the arcade and interrupting someone's roleplay is gross.

**Comment by Elli-Skala** [comment]
Looks great though I'd suggest removing the light replacer and just putting a box of lights or people (me) will just take it every round for the station

**Comment by YakumoChen** [comment]
I still loathe the idea of the escape shuttle and pods docking somewhere where you can just shuttle ride from arrivals. It really kills the immersion of the escape shuttle existing for any real purpose. I'm not sure why we don't just have the shuttle dock be somewhere separate from interlink, but on the same map.

**Comment by restlessObserver** [comment]
> I still loathe the idea of the escape shuttle and pods docking somewhere where you can just shuttle ride from arrivals. It really kills the immersion of the escape shuttle existing for any real purpose. I'm not sure why we don't just have the shuttle dock be somewhere separate from interlink, but on the same map.

Honestly, agreed. I mean, why do we need escape shuttle at all, if we just can fly there in interlink shuttle? I mostly and fully interested in eorg removal only, but if i can have it only in paire with this logical mistake, il go for it. I have no choice.

**Comment by ghost** [comment]
> I still loathe the idea of the escape shuttle and pods docking somewhere where you can just shuttle ride from arrivals. It really kills the immersion of the escape shuttle existing for any real purpose. I'm not sure why we don't just have the shuttle dock be somewhere separate from interlink, but on the same map.

I'll have a solution for this too, want to get SolGov's issue out of the way first though

**Comment by chizzyyyy** [comment]
I don't know.. having the **escape** shuttle dock to somewhere you can be round start takes away from the experience, considering an antag can just do their objectives and afk in interlink
I'm all for a rework for interlink, but removing both eorg and CC as the escape destination? I'm fully against

**Comment by ghost** [comment]
> I don't know.. having the **escape** shuttle dock to somewhere you can be round start takes away from the experience, considering an antag can just do their objectives and afk in interlink I'm all for a rework for interlink, but removing both eorg and CC as the escape destination? I'm fully against

~~regardless of the shuttle the interlink needs this rework~~
The shuttle docking here has its reasoning by the maintainers that requested it, and from the lorebits as well (the interlink being the end of the bluespace hyperlane, so evac docking at cc would just end up flying thru the interlink anyways)

**Comment by Suspect-One** [comment]
I still don't understand why would I go to the emergency shuttle if I can just go to the arrivals and return to the Interlink normally without, well, waiting for the emergency shuttle?

What's so serious about EORG that it requires you to fully rework some areas and think about doing all this if you can just do it better and friendlier to the players?

A) make a small window pop-up in the last 30 seconds of emergency shuttle arriving to the CC. This notification would allow you to teleport to the Interlink or anywhere safe when the round ends. Or just make this whole thing a simple game preference.
B) make everyone immune to damage when the round ends or force peace everyone when the round ends so nobody could get hurt/deal damage. You can also make it a preference thing.

Problem's solved, no EORG unless you have it in your game preferences.

**Comment by ghost** [comment]
hhh
you know what
i'll leave the "eorg removal" code to be done by someone else, this'll just rework the interlink and solgov spawns 🤷

**Comment by Suspect-One** [comment]
Don't get me wrong, I really love the rework itself — Interlink looks exceptionally good now. But I can't understand the whole reasoning behind the emergency shuttle docking at the Interlink. It looks to me like a really bad try following other builds where this is the thing.

Lore-wise (smth about bluespace) it isn't understandable too cause like previously we had Overmap thingy which allowed us to traverse whole sectors of space without anything related to "bluespace hyperlane" and now it's like... gone? How were we doing this in the first place then? How CC sends their ERT via their shuttle straight from CC without the need to board the Interlink? It's so convoluted.

There is only one way to do this if anyone wants this to work. Remove an ability to return to the Interlink via arrivals or the emergency shuttle becomes utterly useless.

p.s. but ofc nobody will do it because there are a lot of mechanics related to the Interlink now.

**Comment by Seris02** [comment]
some things I think should be changed
- dorms looks cold and unappealing because of the floor, and three of the rooms are too small because of a hallway that can't actually be entered, due to my next dot point
- there are centcom fake doors everywhere that are actually indestructible walls
- glass leading to space, afaik the interlink is supposed to be on a planet
- it's missing the lovely garden with trees and stuff!! ):
- the statues next to the shuttle just seem wrong, use plants instead y'know, makes it more endearing and comfortable
- those stairs on the left hand side with the traffic cones in front of them will never ever lead anywhere because the interlink will almost certainly never have more than one z-level
- there appears to be a floating plaque smack dab in the middle of right where you exit the spawn hallway
- please for the love of fuck don't do the one mistake the old interlink did and have monkeys and shit in the glassed off area with trees near the spawn hallway
- the customs gate next to the spawn tram (the small room with the jacket in it) is facing inwards instead of outwards, which looks very awkward
- the hotel pamphlets on the table near the showers don't actually fit on the table

File: tgui/packages/tgui/interfaces/Skyrat_SolGovGTFO.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import { toTitleCase } from 'common/string';
+import { useBackend } from '../backend';
+import { Box, Button, LabeledList, Section, Table, NoticeBox} from '../components';
```
Comment (Seris02) [inline tgui/packages/tgui/interfaces/Skyrat_SolGovGTFO.js]:
```suggestion
import { Box, Button, LabeledList, Section, Table, NoticeBox } from '../components';
```

File: tgui/packages/tgui/interfaces/Skyrat_SolGovGTFO.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            <LabeledList.Item label="Status">
+              {lift_status}
+            </LabeledList.Item>
+            <LabeledList.Item
+              buttons={(
+                <Button
+                  content="Activate Lift"
+                  onClick={() => act('activate_lift')} />
+              )}>
+            </LabeledList.Item>
```
Comment (Seris02) [inline tgui/packages/tgui/interfaces/Skyrat_SolGovGTFO.js]:
```suggestion
            <LabeledList.Item
              buttons={(
                <Button
                  content="Activate Lift"
                  onClick={() => act('activate_lift')} />
              )} />
```

File: tgui/packages/tgui/interfaces/Skyrat_SolGovGTFO.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            </Table.Row>
+            {list_of_riders.map(found_rider => {
+              <Table.Row>
+                <Table.Cell>
+                  {toTitleCase(found_rider.get_authentification_name())}
+                </Table.Cell>
+                <Table.Cell collapsing textAlign="right">
+                  <Box color="label" inline>
+                    {found_rider.get_assignment()}
+                    </Box>
```
Comment (Seris02) [inline tgui/packages/tgui/interfaces/Skyrat_SolGovGTFO.js]:
```suggestion
                  </Box>
```

File: tgui/packages/tgui/interfaces/Skyrat_SolGovGTFO.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+                </Table.Cell>
+              </Table.Row>
+            })};
```
Comment (Seris02) [inline tgui/packages/tgui/interfaces/Skyrat_SolGovGTFO.js]:
```suggestion
            {list_of_riders.map(found_rider => (
              <Table.Row>
                <Table.Cell>
                  {toTitleCase(found_rider.get_authentification_name())}
                </Table.Cell>
                <Table.Cell collapsing textAlign="right">
                  <Box color="label" inline>
                    {found_rider.get_assignment()}
                    </Box>
                </Table.Cell>
              </Table.Row>
            ))}
```
I could be wrong about this particular suggestion but I'm pretty sure this is the correct way to do it.

**Comment by jjpark-kb** [comment]
Just to leave this here: We are having the escape shuttle dock at interlink; this was a decision made already.

We will be working on it to make it required to escape through escape shuttle for "completion."

**Comment by kannthus** [comment]
It looks absolutely amazing, I will admit I'll miss the big garden but all things considered not a big loss for how substantial of an update this is. ❤️

**Comment by ghost** [comment]
> It looks absolutely amazing, I will admit I'll miss the big garden but all things considered not a big loss for how substantial of an update this is. ❤️

that plating hallway against the big window?
Its a biodome, it changes every shift so it doesnt have anything there in the screenshot >:)
Dont worry, the garden will be there. But also a beach, or a snowy wonderland, or a dense jungle.,.,.

**Comment by ghost** [comment]
> some things I think should be changed
> 
>     * dorms looks cold and unappealing because of the floor, and three of the rooms are too small because of a hallway that can't actually be entered, due to my next dot point
> 
>     * there are centcom fake doors everywhere that are actually indestructible walls
> 
>     * glass leading to space, afaik the interlink is supposed to be on a planet
> 
>     * it's missing the lovely garden with trees and stuff!! ):
> 
>     * the statues next to the shuttle just seem wrong, use plants instead y'know, makes it more endearing and comfortable
> 
>     * those stairs on the left hand side with the traffic cones in front of them will never ever lead anywhere because the interlink will almost certainly never have more than one z-level
> 
>     * there appears to be a floating plaque smack dab in the middle of right where you exit the spawn hallway
> 
>     * please for the love of fuck don't do the one mistake the old interlink did and have monkeys and shit in the glassed off area with trees near the spawn hallway
> 
>     * the customs gate next to the spawn tram (the small room with the jacket in it) is facing inwards instead of outwards, which looks very awkward
> 
>     * the hotel pamphlets on the table near the showers don't actually fit on the table

god thank you for actual MAPPING commentary


dorm floors can be fixed
the fakedoors are intentional so it actually has the feel of a station; having geometrically impossible hallways would ruin that feeling, so the 'maintenance' is intentionally empty; making them actual usable maintenance doors IS on the table, though..

interlink being on a planet is dumb, especially the actual lore purpose 'being the end of the hyperspace lane'; that would tear a planet apart

the plating-room to the left of the arcade is an interchangable bio-hall, dont worry

statues felt weird to me too, i'll play with a few things

gonna move those, yea; anywhere with cones is meant to say 'mappers, build off this way if you expand' but those ones do NOT work

plaque is a floor object, not floating

valid, fuck the monkey. i'll put butterflies :)

thats an info booth, but yea i'll swap the windoor

easy fix yehyeh

File: tgui/packages/tgui/interfaces/Skyrat_SolGovGTFO.js (hunk 3 above)
Comment (ghost) [reply tgui/packages/tgui/interfaces/Skyrat_SolGovGTFO.js]: formatting moment hnngh

File: tgui/packages/tgui/interfaces/Skyrat_SolGovGTFO.js (hunk 6 above)
Comment (ghost) [reply tgui/packages/tgui/interfaces/Skyrat_SolGovGTFO.js]: eh, the whole thing needs to be looked back over, the list in its entirety dont work. I'm gonna commit these suggestions though so they fix the formatting a tad bit

**Comment by YakumoChen** [comment]
> Just to leave this here: We are having the escape shuttle dock at interlink; this was a decision made already.
> 
> We will be working on it to make it required to escape through escape shuttle for "completion."

Adding code incentives to take the emergency shuttle does nothing to salvage the immersion lost here. Even if you are not a traitor this move taints all perception of escape.
-shuttle timers no longer feel sensible. Why does the e-shuttle, deployed to remove the crew from potential danger as quickly as possible, take several times longer to reach a spot a dedicated ferry can move in half a minute?
-Does the normal crew feel any need to board a round end shuttle and leave the station when they can take a ferry at their own leisure? You no longer feel stranded if you miss all means to take escape.
-Does the round end destination really provide any safety if a ferry still runs between the station and it?
Escape pods as a concept no longer make any sense since they are now just one way ferries to interlink (launching them "early" to lavaland no longer makes sense either)

**Comment by softcerv** [comment]
EORG should really be kept in as an optional thing. Maybe have like a teleporter to the thunderdome that activates at round end or something.

**Comment by ghost** [comment]
> > Just to leave this here: We are having the escape shuttle dock at interlink; this was a decision made already.
> > We will be working on it to make it required to escape through escape shuttle for "completion."
> 
> Adding code incentives to take the emergency shuttle does nothing to salvage the immersion lost here. Even if you are not a traitor this move taints all perception of escape. -shuttle timers no longer feel sensible. Why does the e-shuttle, deployed to remove the crew from potential danger as quickly as possible, take several times longer to reach a spot a dedicated ferry can move in half a minute? -Does the normal crew feel any need to board a round end shuttle and leave the station when they can take a ferry at their own leisure? You no longer feel stranded if you miss all means to take escape. -Does the round end destination really provide any safety if a ferry still runs between the station and it? Escape pods as a concept no longer make any sense since they are now just one way ferries to interlink (launching them "early" to lavaland no longer makes sense either)

for the second argument, there IS a reason the shuttle takes longer

the ferry uses a Quantum Pull Engine, tied between two set points (the interlink and the station); while this allows for fast travel, its only between two points.
the evac shuttle is equipped for recovery of a large crew from the station, including wounded or convicts; those realistically can't be transferred via the ferry.

and for the second-to-last argument, the future plan should include the ferry 'evacuating' as well on relevant alerts, as it WOULD be a breach of safety, yes.
But, as I've decided, not for this PR, this is just an interlink rework not the end-round rework.

**Comment by thestubborn** [comment]
image
image

**Comment by YakumoChen** [comment]
@OrionTheFox I don't feel there's quite enough dorm rooms. Also there ought to be a water supply for the showers to refill with.

**Comment by ghost** [comment]
> @OrionTheFox I don't feel there's quite enough dorm rooms. Also there ought to be a water supply for the showers to refill with.
hh
got one more dorm, I rarely see even all the large ones full so this should be plenty
I also set the refill_rate on all the showers to 4 (it drains 5 per second and by default recharges 0.5) so it'll last a lot longer

**Comment by thestubborn** [comment]
i'll state the reason for my thumbs down despite
actually really liking this overhaul
I simply disagree with having the escape shuttle dock at the link for reasons outlined above, but anyway

any reason the loot piles have been scrapped?
i think the NT Rep should have the grandfather clock, not the blueshield but that's just my personal view on the role, up to you
image
missing wall *minecraft cave sounds*
image
these railings should probably be a glass door/pane
should probably give the private officers a scary toilet

**Comment by ghost** [comment]
> i'll state the reason for my thumbs down despite actually really liking this overhaul I simply disagree with having the escape shuttle dock at the link for reasons outlined above, but anyway
> 
> any reason the loot piles have been scrapped? i think the NT Rep should have the grandfather clock, not the blueshield but that's just my personal view on the role, up to you image missing wall _minecraft cave sounds_ image these railings should probably be a glass door/pane should probably give the private officers a scary toilet

Understandable, no longer the concern of this PR though, this is simply the mapping half; if that ends up being scrapped, the entire east side can be remodelled.

Loot piles made the Interlink have loot, which was kinda counter-intuitive to the 'safezone'; if a maintainer says they should stay I'll listen, but otherwise its probably for the best the Interlink is properly safe and not used for mechanics besides some food and clothing.
NT Rep already has a display with a badge that they cant open, the clock also surprisingly doesnt have any mechanical time-display to it, so while I understand it I think that the more formal should have the award, and the more strict should have the anxiety-inducing clock: if you end up in the Blueshield office you're in trouble, and the idea of a ticking clock adds to that imo.

I can fix those all, and agree that a security/prison toilet would be silly. Thanks for the review :)


Aaand done

**Comment by ghost** [comment]
>:) the UI works really well now
All thats left is the actual function, and to get solgov spawns working

**Pull request closed without merging**